src/modules/ekf2/EKF/common.h

```diff
@@ -56,26 +56,82 @@ using matrix::Vector2f;
 using matrix::Vector3f;
 using matrix::wrap_pi;
 
-enum class velocity_frame_t : uint8_t {
-	LOCAL_FRAME_FRD,
-	BODY_FRAME_FRD
+// maximum sensor intervals in usec
+#define BARO_MAX_INTERVAL (uint64_t)2e5 ///< Maximum allowable time interval between pressure altitude measurements (uSec)
+#define EV_MAX_INTERVAL   (uint64_t)2e5 ///< Maximum allowable time interval between external vision system measurements (uSec)
+#define GPS_MAX_INTERVAL  (uint64_t)5e5 ///< Maximum allowable time interval between GPS measurements (uSec)
+#define RNG_MAX_INTERVAL  (uint64_t)2e5 ///< Maximum allowable time interval between range finder  measurements (uSec)
+
+// bad accelerometer detection and mitigation
+#define BADACC_PROBATION  (uint64_t)10e6        ///< Period of time that accel data declared bad must continuously pass checks to be declared good again (uSec)
+#define BADACC_BIAS_PNOISE      4.9f    ///< The delta velocity process noise is set to this when accel data is declared bad (m/sec**2)
+
+// ground effect compensation
+#define GNDEFFECT_TIMEOUT       10E6    ///< Maximum period of time that ground effect protection will be active after it was last turned on (uSec)
+
+enum class VelocityFrame : uint8_t {
+	LOCAL_FRAME_FRD = 0,
+	BODY_FRAME_FRD  = 1
+};
+
+enum GeoDeclinationMask : uint8_t {
+	// Bit locations for mag_declination_source
+	USE_GEO_DECL  = (1<<0), ///< set to true to use the declination from the geo library when the GPS position becomes available, set to false to always use the EKF2_MAG_DECL value
+	SAVE_GEO_DECL = (1<<1), ///< set to true to set the EKF2_MAG_DECL parameter to the value returned by the geo library
+	FUSE_DECL     = (1<<2)  ///< set to true if the declination is always fused as an observation to constrain drift when 3-axis fusion is performed
+};
+
+enum MagFuseType : uint8_t {
+	// Integer definitions for mag_fusion_type
+	AUTO    = 0,   	///< The selection of either heading or 3D magnetometer fusion will be automatic
+	HEADING = 1,   	///< Simple yaw angle fusion will always be used. This is less accurate, but less affected by earth field distortions. It should not be used for pitch angles outside the range from -60 to +60 deg
+	MAG_3D  = 2,   	///< Magnetometer 3-axis fusion will always be used. This is more accurate, but more affected by localised earth field distortions
+	UNUSED  = 3,   	///< Not implemented
```

Comment:
Skip entirely?
```suggestion
```

Reply:
Looks like deprecating UNUSED and re-indexing is safe for the remaining logic of INDOOR and NONE.